renku/core/models/datasets.py

```diff
@@ -615,6 +617,130 @@ def __attrs_post_init__(self):
                 self.name, self.version
             )
 
+    @classmethod
+    def from_yaml(cls, path, client=None, commit=None):
```

Comment:
General question, do we still want to use `.yaml` or `.json`? There shouldn't be any performance difference, `.json` is probably better supported with external tools (e.g. you can just copy-paste the content to jsonld playground), if we ever switch to a DB approach (i.e. a key-value store) it'd be json anyways.

Reply:
Will be addressed in https://github.com/SwissDataScienceCenter/renku-python/issues/1299.